rosidl_typesupport_freertps_cpp/rosidl_typesupport_freertps_cpp/__init__.py

```diff
@@ -177,3 +93,72 @@ def generate_typesupport_freertps_cpp(args):
                     minimum_timestamp=latest_target_timestamp)
 
     return 0
+
+
+# TODO(jacquelinekay) use bit shifting/& instead of mod/multiplication!!!
+def enforce_alignment(required_alignment, indent=2):
+    print("""{0}if ((_p - _buf) % {1} != 0) {{
+{0}  _p += (static_cast<uint32_t>(floor((_p - _buf) / {1}) + 1)) * {1} - (_p - _buf);
+{0}}}""".format(' ' * indent, required_alignment))
+
+
+def enforce_read_alignment(required_alignment, indent=2):
+    print("""{0}if ((*_p - initial_p) % {1} != 0) {{
+{0}  uint8_t pad = (static_cast<uint32_t>(
+{0}      floor((*_p - initial_p) / {1})) + 1) * {1} - (*_p - initial_p);
+{0}  if (*_len < pad) {{
+{0}    return false;
+{0}  }}
+{0}  *_p += pad;
+{0}  *_len -= pad;
+{0}}}""".format(' ' * indent, required_alignment))
+
+
+def enforce_alignment_buffer_size(required_alignment, indent=2):
+    print("""{0}if (_len % {1} != 0) {{
+{0}  _len += (static_cast<uint32_t>(floor(_len / {1})) + 1) * {1} - _len;
+{0}}}""".format(' ' * indent, required_alignment))
```

Comment:
Instead of using `{0}` and format to achieve indentation this could call a simple helper function line `print_lines(list_of_lines, indentation)` which would make the generated code much more readable.